src/herder/Upgrades.cpp

```diff
@@ -631,9 +682,16 @@ prepareLiabilities(AbstractLedgerTxn& ltx, LedgerTxnHeader const& header)
 
     auto offersByAccount = ltx.loadAllOffers();
 
-    uint64_t nChangedAccounts = 0;
+    std::unordered_set<AccountID> changedAccounts;
     uint64_t nChangedTrustLines = 0;
+
     std::map<UpdateOfferResult, uint64_t> nUpdatedOffers;
+
+    // We want to keep track of the original minBalances before they are changed
+    // due to sponsorship changes from deleted offers
+    auto offerAccountMinBalanceMap =
```

Comment:
Ended up changing the logic to just store all minBalances. I agree that the `originalMinBalanceMap` logic I had before could be dangerous for future maintainers.